pytorch_lightning/trainer/trainer.py

```diff
@@ -983,7 +983,8 @@ def __load_ckpt_weights(
                     ' specify a path for a checkpoint `.{fn}(ckpt_path=PATH)`'
                 )
 
-            self.training_type_plugin.barrier()
+            if not self._device_type == DeviceType.TPU:
```

Comment:
Can you explain why is it needed ?

Reply:
Please, add a comment to explain why it is needed.

Reply:
Comment added. Thanks.